src/main/kotlin/com/openlattice/mechanic/upgrades/InsertDeletedChronicleEdgeIds.kt

```diff
@@ -0,0 +1,130 @@
+        private val logger = LoggerFactory.getLogger(InsertDeletedChronicleEdgeIds::class.java)
+    }
+
+    override fun upgrade(): Boolean {
+
+        val edgeEntitySet = toolbox.entitySets.getValue(CHRONICLE_EDGES_ENTITY_SET_ID)
+        val partitionsVersion = edgeEntitySet.partitionsVersion
+        val partitions = edgeEntitySet.partitions
+        val partitionsSql = "'{${partitions.joinToString(",")}}'::integer[]"
+
+        val version = System.currentTimeMillis()
+
+        val insertToIdsSql = insertToIdsQuery(partitionsSql, partitionsVersion, version)
+        val insertToDataSql = insertToDataQuery(partitionsSql, partitionsVersion, version)
+
+        logger.info("Insert to ids sql: {}", insertToIdsSql)
+        logger.info("Insert to data sql: {}", insertToDataSql)
+
+
+        logger.info("About to do update")
+        try {
+            toolbox.hds.connection.use { conn ->
+
+                conn.createStatement().use { stmt ->
+                    val sw = Stopwatch.createStarted()
+                    val insertedToIdsCount = stmt.executeUpdate(insertToIdsSql)
+                    logger.info(
+                            "Inserted {} entity key ids into ids table in {} ms.",
+                            insertedToIdsCount,
+                            sw.elapsed(TimeUnit.MILLISECONDS)
+                    )
+
+                    sw.reset().start()
+                    val insertedToDataCount = stmt.executeUpdate(insertToDataSql)
+                    logger.info(
+                            "Inserted {} entity key ids into data table in {} ms.",
+                            insertedToDataCount,
+                            sw.elapsed(TimeUnit.MILLISECONDS)
+                    )
+                }
+            }
+
+            logger.info("Update complete!")
+
+        } catch (e: Exception) {
+            logger.info("Something bad happened :(", e)
+        }
+
+        return true
+    }
+
+    private val withQuery = "WITH chronicle_edge_ids AS (" +
+            "SELECT DISTINCT ${PostgresColumn.EDGE_ENTITY_KEY_ID.name} " +
+            "FROM ${E.name} " +
+            "WHERE ${PostgresColumn.EDGE_ENTITY_SET_ID.name} = '$CHRONICLE_EDGES_ENTITY_SET_ID') "
+
+    private fun insertToIdsQuery(partitions: String, partitionsVersion: Int, version: Long): String {
+
+        val insertCols = listOf(
+                PostgresColumn.PARTITION,
+                PostgresColumn.ENTITY_SET_ID,
+                PostgresColumn.ID_VALUE,
+                PostgresColumn.VERSION,
+                PostgresColumn.VERSIONS,
+                DataTables.LAST_WRITE,
+                PostgresColumn.PARTITIONS_VERSION
+        ).joinToString(",") { it.name }
+
+        return "$withQuery INSERT INTO ${IDS.name} ($insertCols) " +
+                "SELECT " +
+                    "($partitions)[ 1 + (('x'||right(${PostgresColumn.EDGE_ENTITY_KEY_ID.name}::text,8))::bit(32)::int % array_length($partitions,1))] as ${PostgresColumn.PARTITION.name}, " +
```

Comment:
Did you test this expression?

Reply:
yep

Reply:
(btw this is derived from the expression we've used in the past -- just replaced `partitions` with the specific partitions for the chronicle entity set, and replaced `id` with `edge_entity_key_id` (since that's the id column here)